cityofaustin/atd-vz-data
# LDM Blueform crash location trigger

## Associated issues

**FYI: We are still feeling out how to test LDM work so open to any feedback! Going to discuss how this goes in the next LDM standup ~, and this should not get merged down into master branch~. It will be merged into a LDM collector branch ~that is soon to be unblocked~ that is now ready.**

https://github.com/cityofaustin/atd-data-tech/issues/12644

This PR adds the non-CR3 crash location association trigger to the LDM schema.

## Testing
**URL to test:** 
Local only

**Steps to test:**
1. To get a LDM data dump, use the command below. You will need to update the command with the path to your snapshots folder. The snipped part of the hostname and password for the LDM staging DB can be found in 1PW under the entry: `RDS Instance: LDM staging`. The command excludes the change log but will still take a moment.
```bash
PGHOST=ldm-staging.<snip>.us-east-1.rds.amazonaws.com \
PGDATABASE=atd_vz_data \
PGUSER=postgres \
PGPASSWORD=<snip> \
pg_dump \
--clean \
--create \
--if-exists \
--no-owner \
--no-privileges \
--exclude-table-data atd_txdot_change_log \
> /path/to/your/project/atd-vz-data/atd-vzd/snapshots/ldm_<today's date>.sql
```
3. If you can access the VZ read replica, then you should be able run this command. It is restricted to the same IPs as the replica through the security group called `visionzero-rds-read-replica`. We can add anyone who is looking to test this if needed.
4. Once you have the file in your snapshots folder, run `./vision-zero replicate-db -f <your filename>.sql` to restore from the LDM dump that you just made.
5. Connect to your local VZ database and you will know that it worked when you see the `cris` and `vision_zero` schemas.
6. Run the SQL in `atd-vzd/ldm_triggers/update_non_cr3_location/update_noncr3_location.sql`
7. Run the SQL in `atd-vzd/ldm_triggers/update_non_cr3_location/up.sql`
8. You can now run the test steps 1-4 in https://github.com/cityofaustin/atd-vz-data/pull/1269

---
#### Ship list
